src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -54,7 +54,9 @@ internal abstract class AcquireTokenHandlerBase
         protected AcquireTokenHandlerBase(RequestData requestData)
         {
             this.Authenticator = requestData.Authenticator;
-            this.CallState = CreateCallState(this.Authenticator.CorrelationId);
+            this.CallState = CreateCallState(requestData.CorrelationId != Guid.Empty
+                ? requestData.CorrelationId 
+                : this.Authenticator.CorrelationId);
```

Comment:
>this.Authenticator.CorrelationId [](start = 18, length = 32)

Will this work well for Authenticator scenarios? Ideally if the Correlation id is always set to Guild.Empty in all other cases but when we have the decice code flow?

Reply:
Correct, `requestData.CorrelationId` is only set in device code flow, and will be Guid.Empty for all other flows and so therefore the `CallState` will default to using `Authenticator.CorrelationId`.

Reply:
Thanks for explaining. Great for backwards compatibility/consistency. 
The I suppose my only potential concern has nothing to do with your change it is really about the state we set. Do we validate the state is the same as we send?

---
In reply to: [175547615](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1017#discussion_r175547615) [](ancestors = 175547615)

Reply:
It seems that the CallState is not currently validated to make sure it is the same as the one set for any of the flows.